spec/connectors/registry_spec.rb

```diff
@@ -9,21 +9,74 @@
 require 'connectors/registry'
 
 describe Connectors::Factory do
-  let(:factory) do
-    described_class.new
+  subject { described_class.new }
+
+  let(:configuration) {
+    {}
+  }
+
+  let(:job_description) {
+    {}
+  }
+  class MyConnector
+    def initialize(configuration: {}, job_description: {})
+      # nothing needed
+    end
+  end
+
+  before(:each) do
+    subject.register('my-connector', MyConnector)
   end
 
   describe '#connector_class' do
     it 'let us register a connector under a name' do
-      class MyConnector
-        def works
-          'works'
-        end
+      connector_class = subject.connector_class('my-connector')
+      expect(connector_class == (MyConnector)).to eq(true)
+    end
+  end
+
+  describe '#registered?' do
+    context 'my-connector is registered' do
+      it 'should return that sharepoint connector is registered' do
+        expect(subject.registered?('my-connector')).to eq(true)
+      end
+
+      it 'should return that another-connector is not registered' do
+        expect(subject.registered?('another-connector')).to eq(false)
       end
+    end
+  end
+
+  describe '#connector' do
+    context 'my-connector is registered' do
+      it 'should return a my-connector instance' do
+        connector_instance = subject.connector('my-connector', configuration, job_description)
+
+        expect(connector_instance).to_not be_nil
+        expect(connector_instance.is_a?(MyConnector)).to eq(true)
+      end
+    end
+
+    context 'another-connector is not registered' do
```

Comment:
```suggestion
    context 'sharepoint connector has not been registered' do
```

Idk `another-connector` looks like a leftover from a different test. Maybe worth to change `sharepoint` to `another-connector` instead.

Reply:
The message is correct, but I've fixed it by using the `unregistered_connector` field now